hybrid-array/src/lib.rs

```diff
@@ -419,6 +419,25 @@ where
     }
 }
 
+impl<T, U> FromIterator<T> for Array<T, U>
+where
+    U: ArraySize,
+{
+    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
+        let mut iter = iter.into_iter();
+        let ret = Self::from_fn(|_| {
+            iter.next()
+                .expect("iterator should have enough items to fill array")
+        });
+
+        assert!(
+            iter.next().is_none(),
+            "too many items in iterator to fit in array"
```

Comment:
Maybe these should both say "iterator should have exactly as many items as can fit in array"

Reply:
I think it could be useful to know whether you have more than needed or not enough elements in iterator to fill array. The other `expect` message can be "not enough items in iterator to fill array".